accumulo-store/src/main/java/gaffer/accumulostore/key/core/impl/classic/ClassicEdgeDirectedUndirectedFilterIterator.java

```diff
@@ -52,28 +57,56 @@ public boolean accept(final Key key, final Value value) {
                     break;
                 }
             }
-            return !foundDelimiter || checkEdge(rowID);
+            return !foundDelimiter || checkEdge(flag, key);
         }
     }
 
-    private boolean checkEdge(final byte[] rowID) {
-        final byte flag = rowID[rowID.length - 1];
+    private boolean checkEdge(final byte flag, final Key key) {
```

Comment:
Isn't this duplicate code to what's in RangeElementPropertyFilterIterator?  Suggest we extract this out to a utility class to reduce duplication.